File: prueba-tecnica-backend/src/main/java/com/lautaro_cenizo/entity/repuesto/Repuesto.java
package com.lautaro_cenizo.entity.repuesto;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "Repuestos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Repuesto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String codigo;
    private String nombre;
    private String descripcion;
    private BigDecimal precio;
    private Integer stock;
    private Integer stockMinimo;

    @Enumerated(EnumType.STRING)
    private CategoriaRepuesto categoria;
}

File: prueba-tecnica-frontend/src/app/models/class/repuesto.ts
import { CategoriaRepuesto } from "../enums/categoriaRepuesto";

export class Repuesto {
    id!: number;
    codigo!: string;
    nombre!: string;
    descripcion!: string;
    precio!: number;
    stock!: number;
    stockMinimo!: number;
    categoria!: CategoriaRepuesto;
  
    constructor(data?: Partial<Repuesto>) {
      Object.assign(this, data);
    }
  }